autorest/azure/environments.go

```diff
@@ -43,6 +43,7 @@ type Environment struct {
 	ActiveDirectoryEndpoint      string `json:"activeDirectoryEndpoint"`
 	GalleryEndpoint              string `json:"galleryEndpoint"`
 	KeyVaultEndpoint             string `json:"keyVaultEndpoint"`
+	KeyVaultResourceID           string `json:"keyVaultResourceID"`
```

Comment:
since we are here, do you mind add some of these well-known resource IDs too?

| Resource | ResourceID |
|---|---|
| ARMResourceID |  "https://management.azure.com/" |
| GraphResourceID | "https://graph.windows.net/" |
| KeyVaultResourceID | "https://vault.azure.net" |
| DatalakeResourceID  | "https://datalake.azure.net/" |
| MicrosoftGraphResourceID | "https://graph.microsoft.com/" |
| BatchResourceID | "https://batch.core.windows.net/" |
| StorageResourceID | "https://storage.azure.com/" |

Reply:
Looks like existing property 'TokenAudience' is already for 'ARMResourceID'.

I'm fine about adding those, but since there are bunch of items, shall we separate them from existing configs. What about adding a sub property 'TokenAudiences' for all the ids?

e.g.
```
PublicCloud = Environment{
		Name:                         "AzurePublicCloud",
		ManagementPortalURL:          "https://manage.windowsazure.com/",
		PublishSettingsURL:           "https://manage.windowsazure.com/publishsettings/index",
                ...
		TokenAudiences: {
                  ResourceManager = "https://management.azure.com/",
                  Graph = "https://graph.windows.net/",
                  KeyVault= "https://vault.azure.net"
                  ...
                }
	}
```

@yangl900  @jhendrixMSFT

Reply:
Agreed on separating into sub properties.